File: Serenity_Ilk_Hafta_odevi/MultiTenancy/MultiTenancy/MultiTenancy/MultiTenancy.Web/Modules/Administration/Tenant/TenantRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace MultiTenancy.Administration;


[ConnectionKey("Default"), DisplayName("Tenants"),
 InstanceName("Tenant"), TwoLevelCached]
[ReadPermission(PermissionKeys.Tenants)]
[ModifyPermission(PermissionKeys.Tenants)]
[LookupScript("Administration.Tenant")]


public sealed class TenantRow : Row<TenantRow.RowFields>, IIdRow, INameRow{

    [DisplayName("Tenant Id"), Identity]
    public Int32? TenantId
    {
        get => Fields.TenantId[this];
        set => Fields.TenantId[this] = value;
    }

   // [DisplayName("Tenant Id"), Identity, IdProperty]
    //public int? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

//    [DisplayName("Tenant Name"), Size(100), NotNull, QuickSearch, NameProperty]
   // public string TenantName { get => fields.TenantName[this]; set => fields.TenantName[this] = value; }



    public class RowFields : RowFieldsBase
    {
        public Int32Field TenantId;
        public StringField TenantName;

    }
}

File: Serenity_Ilk_Hafta_odevi/MultiTenancy/MultiTenancy/MultiTenancy/MultiTenancy.Web/Modules/ServerTypes/Administration/TenantRow.ts
import { fieldsProxy } from '@serenity-is/corelib';

export interface TenantRow {
    TenantId?: number;
    TenantName?: string;
}

export abstract class TenantRow {
    static readonly idProperty = 'TenantId';
    static readonly nameProperty = 'TenantName';
    static readonly localTextPrefix = 'Administration.Tenant';

    static readonly deletePermission = 'Administration:Tenants';
    static readonly insertPermission = 'Administration:Tenants';
    static readonly readPermission = 'Administration:Tenants';
    static readonly updatePermission = 'Administration:Tenants';

    static readonly Fields = fieldsProxy<TenantRow>();
}